sherpa/plot/__init__.py

```diff
@@ -26,8 +26,30 @@
 and used, even when there is no available plot backend, it is just
 that no graphical display will be created.
 
-"""
+Which backend is used?
+----------------------
+
+When this module is first imported, Sherpa tries to import the
+backends installed with Sherpa in the order listed in
+`sherpa.plot.try_backends`. The first module that imports
+successfully is set as the active backend. The following command prints the
+name and the location on disk of that module::
+
+   >>> from sherpa import plot
+   >>> print(plot.backend)
+
+Change the backend
+------------------
+
+After the initial import, the backend can be changed by loading one of
+the plotting backends shipped with sherpa (or any other module that
+provides the same interface):
 
+  >>> import sherpa.plot.pylab_backend
+  >>> plot.backend = sherpa.plot.pylab_backend
+
+"""
```

Comment:
Is the following now true (text probably needs a re--work)?  If so we should review the docs/ and notebooks. 

Actually, thinking about this, this is only really relevant for users writing "library-style" code rather than someone at a notebook prompt plotting things. Which means the text definitely needs work...

```suggestion

Using the plot classes
----------------------

To allow the backend to be changed, the classes from this module should always be
referenced explicitly with the full path, rather than importing the names directly.
That is:

    >>> from sherpa import plot

and then refernce the objects directly (e.g. `plot.DataPlot`) rather than importing
it directly like

    >>> from sherpa.plot import DataPlot

"""
```

Reply:
It's in the docstring of the module, because that a place where normal users won't see it, but it's likely that future developers (or very advanced users who write their own library-style plot classes) find it.

I plan to make narrative docs with section headings like "for developers: How to implement a plotting backend" and "for developers and advanced users: How to define your own new plot type". However, we already agreed that we'll change the backend modules to backend classes in (my step 1) after this is merged, so I don't want to invest too much time in making detailed docs for something that's going to be out of date literally next week. 

That's one of the drawbacks of chopping the implementation into a series of PRs. I'll reword the text here to that it's not wrong, but I won't make it "good". But please keep that documenation issue in mind when you review the following PRs, because the docs clearly will have to be flashed out more and more over time.